controllers/argocd/route.go

```diff
@@ -392,3 +413,48 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 	}
 	return r.Client.Update(context.TODO(), route)
 }
+
+// The algorithm used by this function is:
+// - If the FIRST label ("console-openshift-console" in the above case) is longer than 63 characters, shorten (truncate the end) it to 63.
+// - If any other label is longer than 63 characters, return an error
+// - After all the labels are 63 characters or less, check the length of the overall hostname:
+//   - If the overall hostname is > 255, then shorten the FIRST label until the host name is < 255
+//   - After the FIRST label has been shortened, if it is < 20, then return an error (this is a sanity test to ensure the label is likely to be unique)
+func shortenHostname(hostname string) (string, error) {
+	if hostname == "" {
+		return "", nil
+	}
+
```

Comment:
@Rizwana777 @jgwest should we add a check here to see if shortening is even required before getting into it? 
Just a check to return the hostname as is if it is already within the size limit

Reply:
yes agreed @jaideepr97 added a check https://github.com/argoproj-labs/argocd-operator/pull/1129/files#diff-34c9d2f4dd86a5067f505bc888a3169f2ff71cca7c59bf83e9f083c91d57b1d6R432 , PTAL

Reply:
@Rizwana777 can this not be a simple 

```
if len(hostname) < maxHostnameLength {
  return hostname, nil
}
```

Reply:
Thanks @Rizwana777 one last little thing for improved readability, I would move this line above the previous line, so we do this check and return before splitting the hostname into labels

Reply:
@jaideepr97  Done, PTAL